src/services/user.js

```diff
@@ -35,11 +37,19 @@ export default class UserService {
 
   static async deleteUser(id) {
     const isUserExist = await findUserById(id);
-    if (isUserExist) {
-      await deleteUser(id);
-    } else {
+    if (!isUserExist) {
       throw new NotFoundError(`User with id = ${id} cannot be deleted`);
     }
+
+    const transaction = await sequelize.transaction();
```

Comment:
It is better to deal with transactions only on data-acess layer. Transactions are not a part of business logic, but service should contain business logic only without coupling it to specific storage type an so on. 

Lets imagine that in the future we will use another kind of a storage where we wont need transactions. In this case our service layer should stay unchanged and we will just replace data-acess layer implementation

Reply:
Done